Ignore old `make_csv_dataset` option when using TensorFlow v1.11

ericricky opened the pull request:
Updated `make_csv_dataset` to not feed the argument `num_parallel_parser_calls` to `tensorflow.contrib.data.make_csv_dataset`  when using TensorFlow version 1.11 and later (see [ these release notes](https://github.com/tensorflow/tensorflow/releases)). Without this update, `make_csv_dataset` throws this error for me:
```
 Error in py_call_impl(callable, dots$args, dots$keywords) :
   TypeError: make_csv_dataset() got an unexpected keyword argument 'num_parallel_parser_calls'
```

jjallaire commented on R/text_line_dataset.R:
```diff
@@ -169,7 +169,8 @@ make_csv_dataset <- function(file_pattern, batch_size,
 
-  tf$contrib$data$make_csv_dataset(
+  ## Gather arguments common to all tensorflow versions
+  opts = list(
```
Our normal convention is to name argument lists as `args` rather than `opts`, could you make this change?

kevinykuo replied:
Please also use `<-` for assignment throughout.

ericricky commented:
@jjallaire @kevinykuo I updated variable names and assignment operators as per your requests. I was also using TensorFlow 1.8 recently and found that `make_csv_dataset` rejected the argument `select_columns`. I added a fix for that to the commit. For future reference, I also found a couple of other arguments, (`comment`, `compression_type`, and `default_float_type`), that weren't common across all TensorFlow versions. These were never implemented in the R interface, and I didn't add them.

@kevinykuo I wasn't sure if you meant for me to use the `<-` assignment operator for assigning elements to lists, as in `args[['select_columns']] <- select_columns` but I made the change anyway. Let me know if you would like me to change back to `=`.
